### PR TITLE
update(userspace/libsinsp): support field transformers in filtercheck data structures

### PR DESCRIPTION
**Pull request by jasondellaluce**
**What type of PR is this?**

/kind design

/kind feature

**Any specific area of the project related to this PR?**

/area libsinsp

/area tests

**Does this PR require a change in the driver versions?**

**What this PR does / why we need it**:

As part of https://github.com/falcosecurity/libs/issues/1789, following up https://github.com/falcosecurity/libs/pull/1793, this contribution supports the new field transformers functionality in the filter data structures evaluated at runtime (i.e. filterchecks).

Note, this does not update nor the filter grammar/language parser, nor changes the compiler that turns ASTs into the runtime-evaluated data structures. This performs the step right before, which is getting the filterchecks classes prepared to be configured with tranformers and field-to-field comparisons once the language will supported (will be contributed subsequently).

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

~500 lines of code come from added tests.

/milestone 0.17.0

**Does this PR introduce a user-facing change?**:

```release-note
update(userspace/libsinsp): support field transformers in filtercheck data structures
```

### CODE REVIEW

File: userspace/libsinsp/sinsp_filtercheck.h [hunk 1]
```diff
@@ -275,9 +370,17 @@ class sinsp_filter_check
...
+	// filter compile time, it populates the filter check values with values extracted
+	// from a right-hand side filter check at runtime.
+	//
+	inline void populate_filter_values_with_rhs_extracted_values(const std::vector<extract_value_t>& values);
+	inline void check_rhs_field_type_consistency() const;
 
-	// used for comparing right-hand single value
-	uint32_t m_val_storage_len;
+	std::list<sinsp_filter_transformer> m_transformers;	
+	std::unique_ptr<sinsp_filter_check> m_rhs_filter_check;
```
Comment (Andreagit97) [inline userspace/libsinsp/sinsp_filtercheck.h:382]:
```suggestion
	std::unique_ptr<sinsp_filter_check> m_rhs_filter_check = nullptr;
```

**Review by Andreagit97 — COMMENTED** [review]
Great job! just one minor nit

**Review by Andreagit97 — APPROVED** [review]
/approve

**Comment by jasondellaluce** [comment]
Seems like the e2e test suite is now failing for things that may be out of our control (cc @therealbobo):
https://github.com/falcosecurity/libs/actions/runs/8783269746/job/24104668438?pr=1795
```
Get:26 https://ppa.launchpadcontent.net/ubuntu-toolchain-r/test/ubuntu jammy/main amd64 libx32quadmath0 amd64 13.1.0-8ubuntu1~22.04 [157 kB]
Fetched 14.4 MB in 6s (2269 kB/s)
E: Failed to fetch mirror+file:/etc/apt/apt-mirrors.txt/pool/main/g/glibc/libc6-dev-i386_2.35-0ubuntu3.6_amd64.deb  404  Not Found [IP: 52.252.75.106 80]
E: Failed to fetch mirror+file:/etc/apt/apt-mirrors.txt/pool/main/g/glibc/libc6-x32_2.35-0ubuntu3.6_amd64.deb  404  Not Found [IP: 52.252.75.106 80]
E: Failed to fetch mirror+file:/etc/apt/apt-mirrors.txt/pool/main/g/glibc/libc6-dev-x32_2.35-0ubuntu3.6_amd64.deb  404  Not Found [IP: 52.252.75.106 80]
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
```

**Comment by therealbobo** [comment]
Maybe I missed an `apt update` and that's the cause 🤔

**Comment by therealbobo** [comment]
This should fix the issue: https://github.com/falcosecurity/libs/pull/1808

**Review by Andreagit97 — APPROVED** [review]
/approve

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**